.github/workflows/syncsheet.yml

```diff
@@ -0,0 +1,72 @@
+name: Sync from Google Sheets
+
+on:
+  workflow_dispatch:
+
+env:
+  TARGET_HEAD_REF: "rewrite"
+  TARGET_REPOSITORY: ${{ github.repository }}
+  FORCE_COLOR: 1
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  sync:
+    name: Sync Data from Google Sheets
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Setup Git Config
+        run: |
+          git config user.name "Github Action"
+          git config user.email "noreply@github.com"
+      - uses: actions/setup-node@v3
+        with:
+          cache: 'npm'
+      - name: Prepare the Branch
+        run: |
+          # Check into the "main" branch (currently rewrite)
+ 
+          git checkout ${TARGET_HEAD_REF}
+          git pull
+
+          # Check if a "chore/sheets" Branch does not exists.
+          if [ -z $(git branch --list --remotes origin/chore/sheets) ]
+          then
+
+            # Create it if it is missing
+            git checkout -b chore/sheets
+            git branch --set-upstream-to=origin/chore/sheets chore/sheets || true
+          fi
+          git checkout chore/sheets
+
+          # This can fail if the branch is not on remote. 
+          # If that's the case, we don't care. It will be created on commit / push
+          git pull || true 
+
+          # If here: Branch exists. But might still have a bad state.
+          # Merge from rewrite and favour their (rewrite) changes on conflicts
+          git rebase origin/${TARGET_HEAD_REF} -X theirs
+          
+          git status
+
+          # If here: Branch set up and will (hopefully :P) not create any issues on changes
+      - name: Install Dependencies from Lock-File
+        run: npm ci
+      - name: Sync with the Sheet
+        run: npx vite-node ./scripts/fetchFromSheet.ts
+        # Sync with the Sheet has created some changes. For the next step these changes have to be staged,
+        # otherwise the diffing will not work
```

Comment:
The diff check is done using `git diff --name-status HEAD`, which compares the staged files with the last commit.

Iirc I had to use this approach (over diffing the unstaged files) because `lint-staged` (as the name implies :P) would otherwise not lint and format the newly generated files.

Here's the relevant segment:

```ts
// scripts/createPrForSheetsIfNeeded.ts

console.log(chalk.blue('Looking for changes since last detection:'));

// This checks the currently staged files.
const localDiff = parseDiffOutput((await $`git diff --name-status HEAD`).stdout).filter(
  // We only care about data-Changes. There *shouldnt* be any other changes, but hey.. you never know :P
  (e) => e.file.startsWith('src/data')
);
console.log(localDiff);
```

Reply:
Changed to 
```yaml
        run: npm ci
      - name: Fetch Changes from Google Sheet
        run: npx vite-node ./scripts/fetchFromSheet.ts
        # Fetching from the Google Sheets might have created some changes in the YAML Files.
        # Stage any YAML Files. The --allow-empty is needed here, otherwise husky will try to 
        # create an empty commit, fail, and as result, break the pipeline.
      - name: Add YAML Changes to Git
        run: |
          git add src/data
          npm run lint:staged -- --allow-empty
          git status
        # Staged changes are used in the createPrForSheetsIfNeeded.ts Script to determine if a new PR and/or commit is needed.
      - name: Run the PR Procedure
```

Reply:
That is more understandable :+1: . Also, I just realized that the misunderstanding was mostly on me (and a little silly). I assumed that the comment was talking about the line above it not the action bellow. I'm not quite sure how I got that notion :D

Reply:
>  I assumed that the comment was talking about the line above it not the action bellow. I'm not quite sure how I got that notion :D

I blame python multiline comments :D